platform/features/table/src/controllers/MCTTableController.js

```diff
@@ -51,22 +61,84 @@ define(
                 self.updateRows($scope.rows);
             };
 
+            /*
+             * Define watches to listen for changes to headers and rows.
+             */
             $scope.$watchCollection('filters', function () {
-                self.updateRows(self.$scope.rows);
+                self.updateRows($scope.rows);
             });
-            $scope.$watchCollection('headers', this.updateHeaders.bind(this));
-            $scope.$watchCollection('rows', this.updateRows.bind(this));
+            $scope.$watch('headers', this.updateHeaders.bind(this));
+            $scope.$watch('rows', this.updateRows.bind(this));
+
+            /*
+             * Listen for rows added individually (eg. for real-time tables)
+             */
+            $scope.$on('addRow', this.newRow.bind(this));
         }
 
         /**
-         * On scroll, calculate which rows indexes are visible and
-         * ensure that an equal number of rows are preloaded for
-         * scrolling in either direction.
+         * If auto-scroll is enabled, this function will scroll to the
+         * bottom of the page
+         * @private
+         */
+        MCTTableController.prototype.scrollToBottom = function() {
+            var self = this;
+
+            //Use timeout to defer execution until next digest when any
+            // pending UI changes have completed, eg. a new row in the table.
+            if (this.$scope.autoScroll) {
+                this.$timeout(function(){
+                        self.scrollable[0].scrollTop = self.scrollable[0].scrollHeight;
+                });
+            }
+        };
+
+        /**
+         * Handles a row add event. Rows can be added as needed using the
+         * `addRow` broadcast event.
+         * @private
+         */
+        MCTTableController.prototype.newRow = function (event, row) {
+            //Add row to the filtered, sorted list of all rows
+            if (this.filterRows([row]).length > 0) {
+                this.insertSorted(this.$scope.displayRows, row);
+            }
+
+            //Keep 'rows' synchronized as it provides the unsorted,
+            // unfiltered model for this view
+            if (!this.$scope.rows) {
+                this.$scope.rows = [];
+            }
+            this.$scope.rows.push(row);
```

Comment:
The initial contract with MCTTable is that you pass `rows` and `headers` to the MCTTable-- the thing using MCTTable provides the data.  Yet, this is confused: when you (the thing using MCTTable) want to add a row to your data, you have to ask the MCTTable to do it.  I would prefer that all modifications to $scope.rows live in the thing that is utilizing the MCTTable, and the `newRow` event only serves to notify the MCTTable that a new row has been added and that it needs to do any work associated with that (e.g. updating filter view or auto-scrolling).  This ensures that data modifications are concerns of the thing using MCTTable and not of the MCTTable-- MCTTable maintains it's own view of the data but doesn't modify that which is provided by the thing that uses it.

My recommendation here would have the event just communicate that a new row was added (it can still pass the row that was added), and Table will do whatever it needs to update it's display state without modifying $scope.rows.